packages/react-mutation-mapper/src/component/column/Signal.tsx

```diff
@@ -46,12 +49,48 @@ export function getSignalData(
         variantAnnotation.signalAnnotation.annotation &&
         variantAnnotation.signalAnnotation.annotation.length > 0
     ) {
-        variantAnnotation.signalAnnotation.annotation.forEach(annotation => {
-            // only have one germline annotation
-            if (annotation.mutationStatus.includes('germline')) {
-                signalData = extendMutations([annotation])[0];
+        // if mutation is somatic OR germline, get annotation depands on mutationType
+        if (variantAnnotation.signalAnnotation.annotation.length === 1) {
+            if (
+                (mutationType === Pathogenicity.BIALLELIC &&
+                    variantAnnotation.signalAnnotation.annotation[0].mutationStatus.includes(
+                        'germline'
+                    )) ||
+                (mutationType === Pathogenicity.SOMATIC &&
+                    variantAnnotation.signalAnnotation.annotation[0].mutationStatus.includes(
+                        'somatic'
+                    ))
+            ) {
+                signalData = extendMutations([
+                    variantAnnotation.signalAnnotation.annotation[0],
+                ]);
             }
-        });
+        }
+        // if mutation is both somatic AND germline, get annotation for both or depands on mutationType
+        else {
+            // if mutationType is undefined, get annotation for both somatic and germline
+            if (mutationType === undefined) {
+                signalData = extendMutations(
+                    variantAnnotation.signalAnnotation.annotation
+                );
+            } else {
+                variantAnnotation.signalAnnotation.annotation.forEach(
+                    annotation => {
+                        if (
+                            mutationType === Pathogenicity.GERMLINE &&
+                            annotation.mutationStatus.includes('germline')
+                        ) {
+                            signalData = extendMutations([annotation]);
```

Comment:
Do we really want to overwrite the existing `signalData` in a for loop? Or should we do `signalData.push(extendMutations([annotation]))` instead?

Reply:
@onursumer Good question, `extendMutations` actrullay could only be called once , probably no need to do `signalData.push(extendMutations([annotation]))`.
The loop iterates every `annotation` from `variantAnnotation.signalAnnotation.annotation`, but there are only 3 different cases that `variantAnnotation.signalAnnotation.annotation` could be:
1. only somatic annotation, variantAnnotation.signalAnnotation.annotation.length() = 1
2. only germline annotation, ariantAnnotation.signalAnnotation.annotation.length() = 1
3. both somatic and germline annotation, ariantAnnotation.signalAnnotation.annotation.length() = 2
We go into this loop only if the data is case 3 and `mutationType` is defined. And `extendMutations` would only be called once, which is when annotation object type matches `mutationType` that we defined